pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_alter_primary_key.go

```diff
@@ -550,9 +551,89 @@ func makeShardedDescriptor(b BuildCtx, t alterPrimaryKeySpec) *catpb.ShardedDesc
 func recreateAllSecondaryIndexes(
 	b BuildCtx, tbl *scpb.Table, newPrimaryIndex, sourcePrimaryIndex *scpb.PrimaryIndex,
 ) {
-	// TODO(postamar): implement in 23.1
-	// Nothing needs to be done because fallBackIfSecondaryIndexExists ensures
-	// that there are no secondary indexes by the time this function is called.
+	publicTableElts := b.QueryByID(tbl.TableID).Filter(publicTargetFilter)
+	// Generate all possible key suffix columns.
+	var inKeySuffix []indexColumnSpec
+	{
+		newPrimaryIndexElts := publicTableElts.Filter(hasIndexIDAttrFilter(newPrimaryIndex.IndexID))
+		scpb.ForEachIndexColumn(newPrimaryIndexElts, func(_ scpb.Status, _ scpb.TargetStatus, ic *scpb.IndexColumn) {
+			if ic.Kind == scpb.IndexColumn_KEY {
```

Comment:
nit: I'd just do `if ic.IndexID == newPrimaryIndex.IndexID && ic.Kind == scpb.IndexColumn_KEY` here instead of having another loop above.

Reply:
Done.